### ISSUE TITLE
should input vector have the same norm ?

### ISSUE THREAD

**Issue by carr123** [dalisson/am_softmax#1]
"so the input to the layer should be a batch X embedding tensor"

hello, should each embedding input have the same norm, so they stay on the same surface of a super-sphere before they input to AMSoftmax layer ?

thanks.

**Comment by dalisson** [comment]
Hello, if you go through the code I wrote the embedding vectors are l2 normalized within the am_softmax layer, the input to this layer shoud just be a batch of embeddings coming out of your network.

**Comment by carr123** [comment]
> Hello, if you go through the code I wrote the embedding vectors are l2 normalized within the am_softmax layer, the input to this layer shoud just be a batch of embeddings coming out of your network.

thanks. 
after L2 norm, all the input vectors will have length 1,  can i enlarge those vectors by multipling a factor say 10, so the classification task will seem easier ?

**Comment by dalisson** [comment]
Honestly, I never tried this, the implementation follows the original paper to compute the am_softmax logits. I am using this am_softmax layer in conjunction with nllloss in a private project inside my company, it works much better than regular logsoftmax for embedding learning.
